api/filters.go

```diff
@@ -414,6 +414,75 @@ func (api *API) addFilterDimension(w http.ResponseWriter, r *http.Request) {
 	api.respond.JSON(ctx, w, http.StatusCreated, resp)
 }
 
+func (api *API) updateFilterDimension(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	filterID := chi.URLParam(r, "id")
+	dimensionName := chi.URLParam(r, "name")
+
+	logData := log.Data{
+		"filter_id":      filterID,
+		"dimension_name": dimensionName,
+	}
+
+	updatedDimension := model.Dimension{
+		Options: []string{},
+	}
+
+	if err := api.ParseRequest(r.Body, &updatedDimension); err != nil {
+		api.respond.Error(
+			ctx,
+			w,
+			http.StatusBadRequest,
+			Error{
+				err:     errors.Wrap(err, "failed to parse update filter request"),
+				logData: logData,
+			},
+		)
+		return
+	}
+
+	// The new dimension won't be present on the dataset (i.e. only `City` will be present, not `Country`),
+	// so instead of validating it against the existing Version, we check to see if the dimension exists in Cantabular.
+	if err := api.validateCantabularDimensionExists(ctx, filterID, updatedDimension.Name); err != nil {
+		api.respond.Error(
+			ctx,
+			w,
+			statusCode(err),
+			Error{
+				err:     errors.Wrapf(err, "error searching for dimension %s in cantabular", updatedDimension.Name),
```

Comment:
We don't want to be exposing the word Cantabular (or any other details of our tech stack) to the public. Also we should be trying to keep error messages static, and add the dynamic data to the log data. Here we already have the dimension name in the so it may not be needed at all.